shared/actions/fs-platform-specific.desktop.js

```diff
@@ -102,10 +108,153 @@ function _open(openPath: string): Promise<*> {
 
 export function openInFileUISaga({payload: {path}}: FsGen.OpenInFileUIPayload, state: TypedState) {
   const openPath = path || Constants.defaultKBFSPath
-  const enabled = state.favorite.fuseStatus && state.favorite.fuseStatus.kextStarted
+  const enabled = state.fs.fuseStatus && state.fs.fuseStatus.kextStarted
   if (isLinux || enabled) {
     return Saga.call(_open, openPath)
   } else {
-    return Saga.sequentially([Saga.put(navigateTo([], [fsTab])), Saga.put(switchTo([fsTab]))])
+    return Saga.put(navigateTo([fsTab, {props: {path: Types.stringToPath(openPath)}, selected: 'folder'}]))
+  }
+}
+
+function waitForMount(attempt: number): Promise<*> {
+  return new Promise((resolve, reject) => {
+    // Read the KBFS path waiting for files to exist, which means it's mounted
+    // TODO: should handle current mount directory
+    fs.readdir(Constants.defaultKBFSPath, (err, files) => {
+      if (!err && files.length > 0) {
+        resolve(true)
+      } else if (attempt > 15) {
+        reject(new Error(`${Constants.defaultKBFSPath} is unavailable. Please try again.`))
+      } else {
+        setTimeout(() => {
+          waitForMount(attempt + 1).then(resolve, reject)
+        }, 1000)
+      }
+    })
+  })
+}
+
+function* waitForMountAndOpenSaga(): Saga.SagaGenerator<any, any> {
+  yield Saga.put(FsGen.createSetFlags({kbfsOpening: true}))
+  try {
+    yield Saga.call(waitForMount, 0)
+    // TODO: should handle current mount directory
+    yield Saga.put(FsGen.createOpenInFileUI({payload: {path: Constants.defaultKBFSPath}}))
+  } finally {
+    yield Saga.put(FsGen.createSetFlags({kbfsOpening: false}))
+  }
+}
+
+export function* installKBFSSaga(): Saga.SagaGenerator<any, any> {
+  const result: RPCTypes.InstallResult = yield Saga.call(RPCTypes.installInstallKBFSRpcPromise)
```

Comment:
could use safeTakeEveryPure for this, just return Saga.sequentially([]) in the 3rd param (actionCreatorsWithResult)

Reply:
I'm not sure I follow...did you want me to trigger using `Saga.put` here?

Reply:
you can do something like
```
// at the bottom of the file
Saga.safeTakeEveryPure(MyInstallAction, installKBFS, installKBFSSuccess)

export const installKBFS = () => Saga.call(RPCTypes.installInstallKBFSRpcPromise)
export const installKBFSSuccess = (result: RPCTypes.InstallResult) => Saga.sequentially([
 Saga.put(FsGen.createInstallKBFSResult({result})),
Saga.put(FsGen.createSetFlags({kbfsOpening: true, kbfsInstalling: false})),
Saga.call(waitForMountAndOpenSaga)
])
```